src/AppBundle/Controller/Ndr/AssetController.php

```diff
@@ -338,18 +338,48 @@ public function summaryAction($ndrId)
 
     /**
      * @Route("/ndr/{ndrId}/assets/{assetId}/delete", name="ndr_asset_delete")
+     * @Template("AppBundle:Common:confirmDelete.html.twig")
      *
      * @return RedirectResponse
      */
     public function deleteAction(Request $request, $ndrId, $assetId)
     {
-        $ndr = $this->getNdrIfNotSubmitted($ndrId, self::$jmsGroups);
+        $form = $this->createForm(FormDir\ConfirmDeleteType::class);
+        $form->handleRequest($request);
+
+        if ($form->isValid()) {
+            $ndr = $this->getNdrIfNotSubmitted($ndrId, self::$jmsGroups);
 
-        if ($ndr->hasAssetWithId($assetId)) {
-            $this->getRestClient()->delete("/ndr/{$ndrId}/asset/{$assetId}");
-            $request->getSession()->getFlashBag()->add('notice', 'Asset removed');
+            if ($ndr->hasAssetWithId($assetId)) {
+                $this->getRestClient()->delete("/ndr/{$ndrId}/asset/{$assetId}");
+                $request->getSession()->getFlashBag()->add('notice', 'Asset removed');
+            }
+
+            return $this->redirect($this->generateUrl('ndr_assets', ['ndrId' => $ndrId]));
         }
 
-        return $this->redirect($this->generateUrl('ndr_assets', ['ndrId' => $ndrId]));
+        $asset = $this->getRestClient()->get("ndr/{$ndrId}/asset/{$assetId}", 'Ndr\\Asset');
+
+        if ($asset instanceof EntityDir\Ndr\AssetProperty) {
+            $summary = [
+                ['label' => 'Type', 'value' => 'Property'],
+                ['label' => 'Address', 'value' => implode(', ', $asset->getAddressValidLines())],
+                ['label' => 'Value', 'value' => $asset->getValue(), 'format' => 'money'],
+            ];
+        } else {
+            $summary = [
+                ['label' => 'Type', 'value' => $asset->getTitle()],
+                ['label' => 'Description', 'value' => $asset->getDescription()],
+                ['label' => 'Value', 'value' => $asset->getValue(), 'format' => 'money'],
+                ['label' => 'Valuation date', 'value' => $asset->getValuationDate(), 'format' => 'date'],
```

Comment:
Can we remove these and just use a <entity>_summary partial or a macro call, this is undoing a lot of our translations

Reply:
Moved to using translations rather than defining strings inside the controller